src/main/java/pl/put/poznan/transformer/logic/domain/dto/ScenarioDTO.java

```diff
@@ -1,11 +1,22 @@
 package pl.put.poznan.transformer.logic.domain.dto;
 
+import org.hibernate.validator.constraints.NotEmpty;
+import pl.put.poznan.transformer.logic.validator.ListOfNonemptyStrings;
+
+import javax.validation.Valid;
 import java.util.ArrayList;
 
 public class ScenarioDTO {
+    @NotEmpty(message = "Title can't be empty")
     private String title;
+
+    @NotEmpty(message = "System actor can't be empty")
     private String systemActor;
-    private ArrayList<String> actors;
+
+    @ListOfNonemptyStrings(message = "Each actor must be a nonempty string")
```

Comment:
Those errors are to specific for the domain to put them in a separate location, we have centralized error handling for everything related to this DTO. If java would allow to put expressions not known at compile time within annotations, then I could use methods to generate empty string error for given field name, but unfortunately it's not possible.